GeoMesa_Feature_List.md

```diff
@@ -0,0 +1,100 @@
+    - Given a set of vectors to search through and a set of vectors to establish proximity, return the members of the former set which lie within the (specified) proximity of members of the latter set
+  - Query against an accumulo GeoMesa store
+    - Pointer: [QueryProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala)
+    - Takes advantage of accumulo optimization to carry out geomesa queries
+  - Find the K nearest neighbors to a given point
+    - Pointer: [KNearestNeighborSearchProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/knn/KNearestNeighborSearchProcess.scala)
+  - Identify unique values for an attribute in results of a CQL query
+    - Pointer: [UniqueProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/unique/UniqueProcess.scala)
+  - Convert points to lines
+    - Pointer: [Point2PointProcess.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-process/src/main/scala/org/locationtech/geomesa/process/Point2PointProcess.scala)
+    - Convert a collection of points into a collection of line segments given a middle term parameter. Optionally break on the day of occurrence. This feature isn't really advertised.
+
+
+## Output
+- geomesa-accumulo
+  - A reader for directly querying a datastore in java/scala
+    - Pointer: [<DataStore>.getFeatureReader](https://github.com/locationtech/geomesa/blob/99d2bcf47a2363f58e05abf7f3c39ef214551ed2/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L354)
+    - This is the best bet for high speed accumulo reads, per the GeoMesa gitter.
+  - Produce a collection of features for a given datastore
+    - Pointer: [<DataStore>.getFeatureSource](https://github.com/locationtech/geomesa/blob/99d2bcf47a2363f58e05abf7f3c39ef214551ed2/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala#L329)
+    - Performance characteristics vs the above reader are unclear. This feature is used, however, in the command line export
+- geomesa-tools (command line tools for interacting with geomesa)
+  - Serialize and export stored features (vectors)
+    - Pointer: [ExportCommand.scala](https://github.com/locationtech/geomesa/blob/b7056fae4988ef524913bf3dc33d9ff2a3476b09/geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/commands/ExportCommand.scala)
+    - Supported export formats: CSV, shapefile, geojson, GML, BIN, Avro
+
+## Other Features
+
```

Comment:
other features we support:
* Google Bigtable backend
* multiple indices (including attribute indices) with a cost-based-optimizer for query planning (in accumulo)
* BLOB store - store large files in a pluggable way with geospatial query capabilities - http://www.geomesa.org/documentation/tutorials/geomesa-blobstore.html
* Native API - store arbitrary java objects with geospatial query capabilities - https://github.com/locationtech/geomesa/tree/master/geomesa-native-api